core/src/main/java/com/datastax/oss/driver/internal/core/session/PoolManager.java

```diff
@@ -314,10 +314,14 @@ private void processStateEvent(NodeStateEvent event) {
       NodeState newState = event.newState;
       if (pending.containsKey(node)) {
         pendingStateEvents.put(node, event);
-      } else if (newState == NodeState.FORCED_DOWN) {
+      } else if (newState == null || newState == NodeState.FORCED_DOWN) {
```

Comment:
So when the state is `null` that means the node has been removed? Can we document this somewhere?

Reply:
Yes, `NodeStateEvent` is a bit bare-bones: added is (null, UNKNOWN) and removed is (null, null). I'll add javadocs to the fields.